### PR TITLE
Make sure we do not add initializers to for in vars

### LINKED ISSUE #1951: `void 0 in` in nested for loop
Similar #1647

```
for (let a of [1,2,3]) {
  for (let b in {x:1}) {
    console.log(b);
  }
}
```

The nested for will become

```
for (var b = void 0 in {x: 1}) {
    console.log(b);
}
```

It can reproduce on http://google.github.io/traceur-compiler/demo/repl.html

### PR DESCRIPTION
**Pull request by arv**
Fixes #1951

### CODE REVIEW

File: src/codegeneration/BlockBindingTransformer.js [hunk 1]
```diff
@@ -557,32 +557,38 @@ export class BlockBindingTransformer extends ParseTreeTransformer {
     finder.visitAny(tree);
     if (!finder.found) {
       // just switch it to var
```
Comment (johnjbarton) [inline src/codegeneration/BlockBindingTransformer.js]: Seems like this comment belongs in the `else` region now.

File: src/codegeneration/BlockBindingTransformer.js [hunk 2]
```diff
@@ -557,32 +557,38 @@ export class BlockBindingTransformer extends ParseTreeTransformer {
...
-                    bindingIdentifier, null, declaration.initializer);
-              }
-            ));
-        initializer = renameAll(renames, initializer);
-
-        tree = loopFactory(initializer, renames, renameAll(renames, tree.body));
-        this.revisitTreeForScopes(tree);
-        tree = func(tree);
-      } else {
+      let callFunc = () => {
```
Comment (johnjbarton) [inline src/codegeneration/BlockBindingTransformer.js]: `applyFunc`?

**Comment by johnjbarton** [comment]
LGTM

File: src/codegeneration/BlockBindingTransformer.js (hunk 1 above)
Comment (arv) [inline src/codegeneration/BlockBindingTransformer.js]: Done.

**Pull request merged**